### PR TITLE
Python Client Library: Improvements to docs

### PR DESCRIPTION
**Pull request by marco-rossi29**
action_probabilities -> actions_probabilities

In rl_sim.py:
- `p` was used both for probability and person: used `person`, and `prob` now
- removed one line functions used only once (e.g., pick_a_random_person, create_context_json)
- renamed req_id to event_id

### CODE REVIEW

**Comment by JohnLangford** [comment]
This pull request **introduces 1 alert** when merging b623dbf95d9a2ea452e484a4717e15cc97e7219e into c17fee45682374a1ad394267ecca46d50f6edeb0 - [view on LGTM.com](https://lgtm.com/projects/g/JohnLangford/vowpal_wabbit/rev/pr-937b8b70f2f68bc76347da6690671d89cdbacf77)

**new alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: reinforcement_learning/examples/python/rl_sim.py [hunk 1]
```diff
@@ -55,32 +54,29 @@ def loop(self):
...
-                p = self.pick_a_random_person()
-                context_features = p.get_features()
+                # random.sample() returns a list
+                person = random.sample(self._people, 1)[0]
+                
+                # create context
+                shared_features = person.get_features()
                 action_features = '"_multi": [ {"topic":"HerbGarden"}, {"topic":"MachineLearning"} ]'
-                context_json = self.create_context_json(context_features, action_features)
-                req_id = str(uuid.uuid4())
```
Comment (jackgerrits) [inline reinforcement_learning/examples/python/rl_sim.py]: Remove uuid import

Comment (marco-rossi29) [reply reinforcement_learning/examples/python/rl_sim.py]: Done thanks!

Comment (jackgerrits) [reply reinforcement_learning/examples/python/rl_sim.py]: Run locally - no issues

**Comment by jackgerrits** [comment]
Run locally - no issues. LGTM

**Comment by JohnLangford** [comment]
Why is appveyor failing here? @jackgerrits can you take a look?

**Comment by jackgerrits** [comment]
It looks as though appveyor is timing out - it has reached the 60 minute maximum for a build. It's not clear what has caused the build to get to this long though

**Comment by JohnLangford** [comment]
Could it be the workaround?

-John

**Comment by jackgerrits** [comment]
It's possible. Before the workaround the openssl-windows package took ~5 mins to build, now it is ~7.5 minutes. It seems we have been very close to the 60 minute cap for a while. Perhaps the build has gotten to a point where environmental factor/vm load push us over the cap?

**Comment by JohnLangford** [comment]
How can we cut down the build time?   That seems good for many reasons.

-John

**Comment by marco-rossi29** [comment]
Agree about cutting down the time. For now, I'm going to restart this build to see if the 60min is consistent.

**Comment by jackgerrits** [comment]
I think the easiest way to bring down the build time for appveyor would be to utilize the build caching feature for all of the vcpkg dependencies that we pull and build every time.

https://www.appveyor.com/docs/build-cache/

**Comment by JohnLangford** [comment]
The build cache sounds super-handy if we can get it right.

-John

**Comment by marco-rossi29** [comment]
AppVeyor build succeeded!

**Comment by JohnLangford** [comment]
Merged, thanks :-)

**Pull request merged**